apps/aecore/src/aec_chain_state.erl

```diff
@@ -972,7 +987,7 @@ update_fraud_info(ForkInfoIn, Node, State) ->
 
 handle_top_block_change(OldTopHash, NewTopDifficulty, Node, Events, State) ->
     case get_top_block_hash(State) of
-        OldTopHash -> State;
+        OldTopHash -> {State, []};
```

Comment:
Small bug: the function returns a tuple of a new state and events. If the change had not changed, no new events.

This module would surely benefit from some `-spec`-s